src/dotnet/commands/dotnet-pack/PackCommand.cs

```diff
@@ -109,10 +119,43 @@ public static int Run(string[] args)
                 msbuildArgs.AddRange(argRoot.Values);
 
                 msbuildArgs.AddRange(cmd.RemainingArguments);
-                return new MSBuildForwardingApp(msbuildArgs).Execute();
+                return 0;
             });
 
-            return cmd.Execute(args);
+            int exitCode = cmd.Execute(args);
+            if (msbuildArgs == null)
+            {
+                throw new CommandCreationException(exitCode);
+            }
+
+            return new PackCommand(msbuildArgs, msbuildPath);
+        }
+
+        public static int Run(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            PackCommand cmd;
+            try
+            {
+                cmd = FromArgs(args);
+            }
+            catch (CommandCreationException e)
+            {
+                return e.ExitCode;
+            }
+
+            return cmd.Execute();
+        }
+
+        public ProcessStartInfo GetProcessStartInfo()
+        {
+            return _forwardingApp.GetProcessStartInfo();
+        }
+
+        public int Execute()
+        {
```

Comment:
This method is repeated in a bunch of places and doesn't rely on internals, so it can be deduplicated by making it an extension method.